src/util/misc/projectStroke/StrokeLineJoinProjections.ts

```diff
@@ -199,32 +199,37 @@ export class StrokeLineJoinProjections extends StrokeProjectionsBase {
    * Calculation: the projections are the points furthest from the vertex in
    * the direction of the X and Y axes after distortion.
    *
-   * @todo TODO:
-   *  - Consider only projections that are inside the beginning and end of the circle segment
-   *
    * @see https://github.com/fabricjs/fabric.js/pull/8344#2-3-2-round-skew
    */
   private projectRoundWithSkew(startCircle: Point, endCircle: Point) {
     const projections: Point[] = [];
 
-    const { skewX, skewY } = this.options;
+    const { skewX, skewY, scaleX, scaleY, strokeUniform } = this.options,
+      shearing = new Point(
+        Math.tan(degreesToRadians(skewX)),
+        Math.tan(degreesToRadians(skewY))
+      );
     // The points furthest from the vertex in the direction of the X and Y axes after distortion
-    const circleRadius = new Point()
-        .scalarAdd(this.strokeProjectionMagnitude)
-        .multiply(this.strokeUniformScalar),
-      newY =
-        circleRadius.y / Math.sqrt(1 + Math.tan(degreesToRadians(skewY)) ** 2),
+    const circleRadius = this.strokeProjectionMagnitude,
+      newY = strokeUniform
+        ? circleRadius /
+          scaleY /
+          Math.sqrt(1 / scaleY ** 2 + (1 / scaleX ** 2) * shearing.y ** 2)
+        : circleRadius / Math.sqrt(1 + shearing.y ** 2),
       furthestY = new Point(
-        Math.sqrt(
-          circleRadius.x ** 2 - ((newY * circleRadius.x) / circleRadius.y) ** 2
-        ),
+        // Safe guard due to floating point precision. In some situations the square root
+        // was returning NaN because of a negative number close to zero.
+        Math.sqrt(circleRadius ** 2 - newY ** 2) || 0,
```

Comment:
is this safeguard good enough?
`Math.max` perhaps?

Reply:
I think so. `Math.max(NaN, 0)` and `Math.Min(NaN, 0)` return `NaN` in both cases.

Reply:
I meant inside the sqrt

Reply:
Do you think it's safer? I have no idea..

Reply:
If nan is caused due to sqrt of a negative number then sure, it will solve iy entirely

Reply:
Anyways there is a util ifNaN